### PR TITLE
Use DialogFragment for Android modals

### LINKED ISSUE #22587: [Android] Convert Modal Implementation over to using Dialog Fragment
### Description

Right now we use a somewhat awkward mechanism to do modals in android. We basically add a view to the decor view on top of everything. This requires a ton of custom code to make sure everything operates successfully. 

I'm fairly sure if we just switched to Dialog Fragments we could remove most of that code and then it would all just work more naturally. 

I have a branch [here](https://github.com/dotnet/maui/tree/dialog_fragment)

That mostly works but just needs a bit more testing and cleanup

### PR DESCRIPTION
**Pull request by pictos**
### Description of Change

This should be merged on `net9` branch, but since it's hard to make the .net9-preview don't mess-up with my env. I'm target .net8, when it gets approved this should rebased and re-target. I hope that @PureWeen can do this hard task for me xP.

### Issues Fixed

Fixes #22587

### CODE REVIEW

**Comment by jsuarezruiz** [comment]
Don't worry @pictos, we can retarget to net9.0.

**Comment by PureWeen** [comment]
Android UITests failing
image


The controls device tests are crashing.
I'm guessing the modal tests inside device tests will need to be adjusted somewhat to handle DialogFragments

**Comment by pictos** [comment]
@PureWeen, not sure if all tests will run again, but I think that I fixed the issue. I was using the `DialogFragment.ShowNow` method and it was causing issues with the `FragmentManager`, changing it to `DialogFragment.Show` seems to fix the issue.

**Comment by pictos** [comment]
from the logs looks like the Appium couldn't find the element, but it navigates to the correct screen (at least locally), is that some Appium issue?


> Failed AppDoesntCrashWhenReusingSameTitleView [16 s]
  Error Message:
   System.TimeoutException : Timed out waiting for element...
  Stack Trace:
     at UITest.Appium.HelperExtensions.Wait(Func`1 query, Func`2 satisfactory, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 804
   at UITest.Appium.HelperExtensions.WaitForAtLeastOne(Func`1 query, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 821
   at UITest.Appium.HelperExtensions.WaitForElement(IApp app, String marked, String timeoutMessage, Nullable`1 timeout, Nullable`1 retryFrequency, Nullable`1 postTimeout) in /_/src/TestUtils/src/UITest.Appium/HelperExtensions.cs:line 316
   at Microsoft.Maui.TestCases.Tests.Issues.Issue16499.AppDoesntCrashWhenReusingSameTitleView() in /_/src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue16499.cs:line 19
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)

**Review by PureWeen — CHANGES_REQUESTED** [review]
These two tests fail for me locally

Maybe resolving them will fix the other exceptions?

image

File: src/Core/src/Platform/Android/Resources/anim/nav_modal_default_enter_anim.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<set xmlns:android="http://schemas.android.com/apk/res/android">
+  <translate
+    android:duration="500"
```
Comment (PureWeen) [inline src/Core/src/Platform/Android/Resources/anim/nav_modal_default_enter_anim.xml:4]:
```suggestion
    android:duration="300"
```

File: src/Core/src/Platform/Android/Resources/anim/nav_modal_default_exit_anim.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<set xmlns:android="http://schemas.android.com/apk/res/android">
+  <translate
+      android:duration="500"
```
Comment (PureWeen) [inline src/Core/src/Platform/Android/Resources/anim/nav_modal_default_exit_anim.xml:4]:
```suggestion
      android:duration="300"
```

**Review by PureWeen — APPROVED** [review]

**Comment by PureWeen** [comment]
Test failures are on iOS. This PR only has changes on Android

**Pull request merged**